src/Ocelot/Security/SecurityPolicyExtensions.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Primitives;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Ocelot.Security;
+
+public static class SecurityPolicyExtensions
+{
+    public static string GetClientIpAddress(this HttpContext httpContext, bool tryUseXForwardHeader = true)
+    {
+        if (httpContext == null)
+        {
+            return null;
+        }
+
+        string ip = null;
+
+        // X-Forwarded-For =>  Using the First entry in the list
+        if (string.IsNullOrWhiteSpace(ip) && tryUseXForwardHeader)
+        {
+            ip = httpContext.GetHeaderValue("X-Forwarded-For").SplitCsv().FirstOrDefault();
+        }
+
+        // RemoteIpAddress is always null in DNX RC1 Update1 (bug).
+        if (string.IsNullOrWhiteSpace(ip) && httpContext.Connection?.RemoteIpAddress != null)
+        {
+            ip = httpContext.Connection.RemoteIpAddress.ToString();
+        }
+
+        if (string.IsNullOrWhiteSpace(ip))
+        {
+            ip = httpContext.GetHeaderValue("REMOTE_ADDR");
+        } 
+
+        if (ip == "::1")
+        {
+            ip = "127.0.0.1";
+        }
+
+        return ip;
+    }
+
+    public static string GetHeaderValue(this HttpContext httpContext, string headerName)
+    {
+        if (httpContext?.Request?.Headers?.TryGetValue(headerName, out StringValues values) ?? false)
+        {
+            return values.ToString();
+        }
+
+        return string.Empty;
+    }
+
+    public static List<string> SplitCsv(this string csvList, bool nullOrWhitespaceInputReturnsNull = false)
```

Comment:
```suggestion
    private static IReadOnlyCollection<string> SplitCsv(this string csvList, bool nullOrWhitespaceInputReturnsNull = false)
```

Reply:
Ensuring about strong safety by `IReadOnlyCollection` and making the method `private` don't give much outcome by having it as read-only.

Reply:
this method is only use in this class, so, it must be private. There will always be time to make it public, if necessary